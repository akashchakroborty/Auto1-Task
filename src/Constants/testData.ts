import { CarsResponse } from '../redux/types/carsTypes';

export const MOCK_CARS_DATA: CarsResponse = {
  cars: [
    {
      stockNumber: 97074,
      manufacturerName: 'Porsche',
      modelName: '718 Cayman',
      color: 'black',
      mileage: { number: 100056, unit: 'km' },
      fuelType: 'Diesel',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 12876,
      manufacturerName: 'Tesla',
      modelName: 'Roadster',
      color: 'yellow',
      mileage: { number: 100481, unit: 'km' },
      fuelType: 'Petrol',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 18029,
      manufacturerName: 'Fiat',
      modelName: '131',
      color: 'white',
      mileage: { number: 100710, unit: 'km' },
      fuelType: 'Petrol',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 65672,
      manufacturerName: 'Skoda',
      modelName: 'Felicia',
      color: 'black',
      mileage: { number: 100751, unit: 'km' },
      fuelType: 'Petrol',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 42578,
      manufacturerName: 'Mercedes-Benz',
      modelName: 'Sprinter',
      color: 'yellow',
      mileage: { number: 100782, unit: 'km' },
      fuelType: 'Diesel',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 70917,
      manufacturerName: 'Tesla',
      modelName: 'Roadster',
      color: 'green',
      mileage: { number: 100929, unit: 'km' },
      fuelType: 'Diesel',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 53029,
      manufacturerName: 'Tesla',
      modelName: 'Model X',
      color: 'black',
      mileage: { number: 100997, unit: 'km' },
      fuelType: 'Diesel',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 77489,
      manufacturerName: 'Chrysler',
      modelName: 'Viper',
      color: 'red',
      mileage: { number: 101001, unit: 'km' },
      fuelType: 'Petrol',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 98739,
      manufacturerName: 'Tesla',
      modelName: 'Roadster',
      color: 'blue',
      mileage: { number: 101066, unit: 'km' },
      fuelType: 'Petrol',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
    {
      stockNumber: 72351,
      manufacturerName: 'Porsche',
      modelName: '968',
      color: 'green',
      mileage: { number: 101069, unit: 'km' },
      fuelType: 'Diesel',
      pictureUrl: 'https://auto1-js-task-api--mufasa71.repl.co/images/car.svg',
    },
  ],
  totalPageCount: 100,
  totalCarsCount: 1000,
};

export const MOCK_COLORS = {
  colors: ['red', 'blue', 'green', 'black', 'yellow', 'white', 'silver'],
};

export const MOCK_MANUFACTURERS = {
  manufacturers: [
    {
      name: 'Audi',
      models: [
        {
          name: '100',
        },
      ],
    },
  ],
};
