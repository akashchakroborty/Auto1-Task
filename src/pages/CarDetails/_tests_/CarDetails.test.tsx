import React from 'react';
import { shallow } from 'enzyme';
import { CarDetialsProps, CarDetail } from '..';
import { MOCK_CARS_DATA } from '../../../Constants/testData';

describe('should test the CarDetail Component', () => {
  const setFavorites = jest.fn();
  const mockProps: CarDetialsProps = {
    getCarDetails: jest.fn(),
    clearCarDetails: jest.fn(),
    carDetails: {
      loading: false,
      error: false,
      data: MOCK_CARS_DATA.cars[0],
    },
    match: {
      params: { id: '97074' },
    },
  };
  beforeEach(() => {
    jest.spyOn(React, 'useEffect').mockImplementation((f) => f());
    React.useState = jest.fn().mockReturnValue([[], () => {}]);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('should render the component', () => {
    const wrapper = shallow(<CarDetail {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should test handleFavoriteButtonClick function to save', () => {
    React.useState = jest.fn().mockReturnValue([[], setFavorites]);
    const wrapper = shallow(<CarDetail {...mockProps} />);
    wrapper.find('#favoriteButton').simulate('click');
    expect(setFavorites).toHaveBeenCalled();
  });
  test('should test handleFavoriteButtonClick function to remove', () => {
    React.useState = jest.fn().mockReturnValue([[97074], setFavorites]);
    const wrapper = shallow(<CarDetail {...mockProps} />);
    wrapper.find('#favoriteButton').simulate('click');
    expect(setFavorites).toHaveBeenCalled();
  });
});
