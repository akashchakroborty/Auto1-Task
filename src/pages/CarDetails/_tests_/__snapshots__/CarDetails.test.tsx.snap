// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the CarDetail Component should render the component 1`] = `
<div>
  <div
    className="makeStyles-image-2"
  >
    <ForwardRef(CardMedia)
      className="makeStyles-image-2"
      image="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
    />
  </div>
  <ForwardRef(Container)
    className="makeStyles-container-1"
    maxWidth="md"
  >
    <ForwardRef(Grid)
      container={true}
      spacing={4}
    >
      <ForwardRef(Grid)
        item={true}
        md={8}
        sm={12}
        xs={12}
      >
        <ForwardRef(Typography)
          component="h1"
          fontWeight={700}
          variant="h4"
        >
          Porsche
           
          718 Cayman
        </ForwardRef(Typography)>
        <ForwardRef(Typography)
          component="h2"
          style={
            Object {
              "margin": "24px 0",
              "textTransform": "capitalize",
            }
          }
          variant="h6"
        >
          Stock #
           
          97074
           
          -
           
          100,056
           
          KM
           
          -
           
          Diesel
           
          -
          black
        </ForwardRef(Typography)>
        <ForwardRef(Typography)>
          This car is currently available and can be delivered as soon as tomorrow morning. Please be aware that delivery times shown in this page are not definitive and may change due to bad weather conditions.
        </ForwardRef(Typography)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        md={4}
        sm={12}
        xs={12}
      >
        <ForwardRef(Paper)
          className="makeStyles-favoriteContainer-3"
          square={true}
          variant="outlined"
        >
          <ForwardRef(Typography)
            align="left"
            gutterBottom={true}
          >
            If you like this car, click the button and save it in your collection of favorite items.
          </ForwardRef(Typography)>
          <ForwardRef(Button)
            className="makeStyles-favoriteButton-4"
            color="primary"
            fullWidth={false}
            id="favoriteButton"
            onClick={[Function]}
            size="medium"
            variant="contained"
          >
            Save
          </ForwardRef(Button)>
        </ForwardRef(Paper)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(Container)>
</div>
`;
